src/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.cs

```diff
@@ -270,6 +270,11 @@ public static FileVersionInfo GetVersionInfo(string fileName)
             {
                 throw new FileNotFoundException(fileName);
             }
+            // Check if fileName is a full path.
+            if (!Path.IsPathRooted(fileName))
```

Comment:
I do not think that this is the right condition to use. IsPathRooted returns true for `c:kernel32.dll` - that still gets resolved against current directory for drive `c:`. I think that the right condition to use here would be IsPathFullyQualified.